lib/oc-tests/test_dirDepthUpload.py

```diff
@@ -0,0 +1,190 @@
+numFilesToCreate = config.get('test_numFilesToCreate', 10)
+dir_depth = config.get ('dir_depth',5)
+num_users = config.get ('oc_number_test_users', 3)
+num_file_rows = config.get ('file_row_count', 100)
+style = config.get ('dir_depth_style', 'complex')
+test_with_mover = True
+
+testsets = [
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':5,
+        'test_numFilesToCreate':500,
+        'test_filesizeKB':2,
+        'style': 'flat',
+    },
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':5,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':200,
+        'style': 'complex',
+    },
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':10,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':2000,
+        'sytle': 'flat'
+    },
+    {
+        'num_file_rows':1000,
+        'num_users':10,
+        'dir_depth':10,
+        'test_numFilesToCreate':50,
+        'test_filesizeKB':2000,
+        'sytle': 'complex'
+    },
+]
+
+@add_worker
+def setup(step):
+
+    step (1, 'Create test users')
+    reset_owncloud_account(num_test_users=num_users)
+    check_users(num_users)
+
+    reset_rundir()
+    reset_server_log_file()
+
+    step (6, 'Validate server log file is clean')
+
+    d = make_workdir()
+    scrape_log_file(d)
+
+
+def uploader (step):
+
+    step (2,'Create workdir')
+    d = make_workdir()
+    uploader_num = get_user_number_from_work_directory(d)
+
+    step (3,'Create directories and files then sync and validate')
+
+    procName = reflection.getProcessName()
+
+    if style is 'flat':
+        for i in range(dir_depth):
+            dir_name = "%s/%s_%d"%(procName, 'upload_dir', i)
+            upload_dir = make_workdir(dir_name)
+            for j in range(0, numFilesToCreate):
+                filename = "%s%s%i%s" % (upload_dir,'/TEST_FILE_NEW_USER_SHARE_',j,'.dat')
+                createfile(os.path.join(d,filename),'0',count=1000,bs=filesizeKB)
+    else:
+        dir_name = procName
+        for i in range(dir_depth):
+            dir_name = "%s/%s_%d"%(dir_name, 'upload_dir', i)
+            upload_dir = make_workdir(dir_name)
+            for j in range(0, numFilesToCreate):
+                filename = "%s%s%i%s" % (upload_dir,'/TEST_FILE_NEW_USER_SHARE_',j,'.dat')
+                createfile(os.path.join(d,filename),'0',count=1000,bs=filesizeKB)
+
+    step (4, 'Sync and validate files')
+
+    time_string = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime())
+    logger.info ('time before sync start is %s' % time_string)
+
+    run_ocsync(d,user_num=uploader_num)
+
+    time_string = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime())
+    logger.info ('time after sync start is %s' % time_string)
+
+    checkFilesExist(d) 
+
+    step (5, 'Uploader final step')
+
+for i in range(num_users):
+    add_worker (uploader,name="uploader%02d"%(i+1))
+
+def mover (step):
```

Comment:
What I see is that the method just do a sleep, I guess to introduce a delay. If this is the case, why spawn several processes to add a delay? why not introduce the delay (conditionally or not) in the respective worker?